src/base.ts

```diff
@@ -132,36 +114,79 @@ export const fetchPath = async (
   }
 }
 
-export const postData = async (
-  service: Service,
-  path: string,
+const buildOptions = ({
+  options,
+  method,
+}: {
+  options: RequestOptionsWithRecaptcha
+  method: Method
+}): RequestOptionsWithMethod => {
+  const { recaptchaToken, headers, ...otherOptions } = options
+
+  return {
+    ...otherOptions,
+    headers: {
+      ...headers,
+      ...(recaptchaToken ? { "Recaptcha-Token": recaptchaToken } : {}),
+    },
+    method,
+  }
```

Comment:
there is some header merging going on in `fetchPath` as well. worth to harmonize or a completely different thing?

Reply:
Oh, good point. We can move it to the top-level fetch call and we don't need to worry about recaptcha token anywhere.